### PR TITLE
add configmap init and watcher

### PR DESCRIPTION
**Pull request by AjayTripathy**

### CODE REVIEW

**Comment by dwbrown2** [comment]
Would you be able to add a test to this?

File: costmodel/router.go [hunk 1]
```diff
@@ -822,6 +823,24 @@ func Initialize() {
...
+	watchConfigFunc := func(c interface{}) {
+		conf := c.(*v1.ConfigMap)
+		if conf.GetName() == "pricing-configs" {
+			_, err := cloudProvider.UpdateConfigFromConfigMap(conf.Data)
+			if err != nil {
+				klog.Infof("ERROR UPDATING CONFIG: %s", err.Error())
+			}
+		}
+	}
+	// We need an initial invocation because the init of the cache has happened before we had access to the provider.
```
Comment (mbolt35) [inline costmodel/router.go]: 😢

File: costmodel/router.go [hunk 2]
```diff
@@ -822,6 +823,24 @@ func Initialize() {
...
+	configs, err := kubeClientset.CoreV1().ConfigMaps("kubecost").Get("pricing-configs", metav1.GetOptions{})
```
Comment (mbolt35) [inline costmodel/router.go]: Saw in the clustercache we used `os.Getenv("KUBECOST_NAMESPACE")` - should that be the same here? I'm guilty of this in a few places I think.

Comment (AjayTripathy) [reply costmodel/router.go]: Yes thank you good catch.

**Pull request merged**